app/services/sync_provider_from_find.rb

```diff
@@ -0,0 +1,38 @@
+class SyncProviderFromFind
+  def self.call(provider_code:)
+    find_provider = FindAPI::Provider
+      .current_cycle
+      .includes(:sites, courses: [:sites])
+      .find(provider_code)
+      .first
+    find_courses = find_provider.courses
+    find_sites = find_provider.sites
+
+    provider = Provider.find_or_create_by(code: provider_code)
+    provider.name = find_provider.provider_name
+    provider.save
+
+    sites = find_sites.map do |find_site|
+      site = provider.sites.find_or_create_by(code: find_site.location_name)
+      site.name = find_site.location_name
+      site.save
+      site
+    end
+
+    find_courses.each do |find_course|
+      course = provider.courses.find_or_create_by(code: find_course.course_code)
+      course.name = find_course.name
+      course.level = find_course.level
+      course.start_date = Date.parse(find_course.start_date)
+      course.save
+
+      find_course.sites.each do |find_course_site|
+        CourseOption.find_or_create_by(
+          site_id: sites.find { |s| s.name == find_course_site.location_name }.id,
+          course_id: course.id,
+          vacancy_status: 'B', # TODO: Should this be reflected by `find_course.has_vacancies?`
+        )
+      end
+    end
+  end
+end
```

Comment:
this method is a bit hard to follow and to maintain, it would be nice to simplify it with some private and well named methods.